synapse/replication/http/federation.py

```diff
@@ -248,14 +263,16 @@ def __init__(self, hs: "HomeServer"):
         self.store = hs.get_datastore()
 
     @staticmethod
-    async def _serialize_payload(room_id, args):
```

Comment:
The only caller of this seems to only pass a `room_id`, I'm unsure if this is fixing a bug or not:

https://github.com/matrix-org/synapse/blob/251b5567ecc8fb3d6debaa3f77f6ec2620877d36/synapse/handlers/federation.py#L1379-L1389

Reply:
It seems to be called via `do_invite_join` which is definitely called, it seems to eventually come from either `RoomMemberMaster` or `RoomMemberWorker`, maybe we're not routing anything that calls the worker version to the worker?

Reply:
I think there is a clue here:

https://github.com/matrix-org/synapse/blob/251b5567ecc8fb3d6debaa3f77f6ec2620877d36/synapse/handlers/federation.py#L452-L474

I couldn't see any other call site of `_clean_room_for_join`. So yeah, I think this is never called on workers as you say. Given that, let's go ahead with the fix as written. (Strictly speaking it probably ought to be a separate change, but I don't mind sneaking it in here.)